gateway/modules/crud/operations.go

```diff
@@ -53,7 +54,15 @@ func (m *Module) Read(ctx context.Context, dbAlias, project, col string, req *mo
 	if err := crud.IsClientSafe(); err != nil {
 		return nil, err
 	}
-
+	if req.IsBatch {
+		key := model.ReadRequestKey{DBType: dbAlias, Col: col, HasOptions: req.Options.HasOptions, Req: *req}
+		dataLoader, ok := m.getLoader(fmt.Sprintf("%s-%s-%s", project, dbAlias, col))
+		if !ok {
+			dataLoader = m.createLoader(fmt.Sprintf("%s-%s-%s", project, dbAlias, col))
+		}
+		defer dataLoader.Clear(ctx, key) // clear the key
```

Comment:
We shouldn't do this here. Create a new loader in set config. No need to clear the keys at all